contribution/development_guide.md

```diff
@@ -105,7 +160,54 @@
 
       You could skip vagrant step completely if you're directly compiling Kubearmor on any Linux distro, or using Virtualbox.
 
-      Please ensure that the steps to setup K8s is followed so as to resolve any open dependencies.
+      Please ensure that the steps to setup K8s are followed so as to resolve any open dependencies.
+
+2. Self-managed Kubernetes
+   * Requirements
+
+     Here is the list of minimum requirements for self-managed Kubernetes.
+
+     ```text
+     OS - Ubuntu 18.04
+     Kubernetes - v1.19
+     Docker - 18.09 or Containerd - 1.3.7
+     Linux Kernel - v4.15
+     LSM - AppArmor
+     ```
+
+     KubeArmor is designed for Kubernetes, which means that Kubernetes should be ready in your environment. If Kubernetes is not prepared yet, please refer to [Kubernetes installation guide](self-managed-k8s/README.md). KubeArmor also requires either Docker or Containerd since it internally uses its APIs. KubeArmor requires LSMs to operate properly; thus, please make sure that your environment supports LSMs \(at least, AppArmor\). Otherwise, KubeArmor will work as Audit-Mode with no container behavior restriction.
+
+      * Alternative Setup - Minikube
+
+        MiniKube does not support LSMs by default; thus, you cannot test KubeArmor on Minikube. However, we provide the custom ISO image for testing KubeArmor on Minikube.  
+        Please follow the instructions in [Minikube installation guide](minikube/README.md).
+
+      * Alternative Setup - K3s
+
+        You can also develop and test KubeArmor on K3s instead of the self-managed Kubernetes.  
+        Please follow the instructions in [K3s installation guide](k3s/README.md).
+
+      * Alternative Setup - MicroK8s
+
+        You can also develop and test KubeArmor on MicroK8s instead of the self-managed Kubernetes.  
+        Please follow the instructions in [MicroK8s installation guide](microk8s/README.md).
+
+      * No Support - Docker Desktops
+
+        KubeArmor does not work with Docker Desktops on Windows and macOS because KubeArmor integrates with Linux-kernel native primitives (including LSMs).
+
+   * Development Setup
+
+     In order to install all dependencies, please run the following command.
+
+     ```text
+     $ cd KubeArmor/contribution/self-managed-k8s
+     ~/KubeArmor/contribution/self-managed-k8s$ ./setup.sh
+     ```
+
+     [setup.sh](self-managed-k8s/setup.sh) will automatically install BCC, Go, Protobuf, and some other dependencies.
+
```

Comment:
We should explicitly link to installation guides of BCC, Go.... in case someone wants to setup in an environment not supported in our script